### PR TITLE
added more operator specific dependencies

### PR DESCRIPTION
**Pull request by xh3b4sd**
For instance in the KVM operator there are still dependecies hard coded. See https://github.com/giantswarm/kvm-operator/blob/17e4d4d2d52a4a9bba290ae0c3ef94ee581831b2/resources/master.go#L78. Their docker images might not change frequently, but depending on installations we have to change the endpoint of the docker registry. Thus I plan to make these dependencies injectable. PRs in kubernetesd and the several operators and TPRs will follow.

### CODE REVIEW

File: operator/endpointupdater/docker/docker.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package docker
+
+type Docker struct {
+	Image string
```
Comment (xh3b4sd) [inline operator/endpointupdater/docker/docker.go]: I realized that certain structures in the clustertpr still miss JSON and YAML tags. Will fix this in the next PR separately.

**Review by rossf7 — COMMENTED** [review]
I like that we're moving this to the TPR.

My only query is in the operator block will some images like the bridge be KVM only? For AWS do we just not populate them? I still have some gaps in my understanding of the KVM setup.

**Comment by xh3b4sd** [comment]
I thought about this separation as well. Uncertain. This operator block was probably a bad idea. Should we move the KVM specifics into the KVM TPR?

**Comment by rossf7** [comment]
I feel like it should be in the kvmtpr the same way we have AWS specific stuff in awstpr. But I'm fine with continuing and raising an issue for that.

**Review by rossf7 — APPROVED** [review]

**Comment by xh3b4sd** [comment]
I will put the KVM specifics into the KVM TPR.

**Comment by xh3b4sd** [comment]
@rossf7 please have another look. I removed the operator block in this PR. Will prepare other PRs for the KVM TPR.

**Comment by rossf7** [comment]
LGTM. I like a lot that we're moving these to kvmtpr.

**Comment by xh3b4sd** [comment]
Towards https://github.com/giantswarm/giantswarm/issues/1428.

**Pull request merged**